ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
I'd say we need to find a way of giving the CNCI the appropriate amount of memory for both SingleVM and real clusters as part of this PR.  If we commit without a solution to this we'll either break Single VM (for people with less powerful computers) or we break ciao on real clusters.

#912 is important but I would say doesn't need to be done in this PR as it is an existing problem.